core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -92,6 +92,16 @@ default ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSess
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support analyze");
     }
 
+    /**
+     * Returns a table handle for the specified table name, or null if the connector does not contain the table.
+     * The returned table handle can contain information in tupleDomain.
+     */
+    @Nullable
+    default ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSession session, SchemaTableName tableName, TupleDomain<ColumnHandle> tupleDomain)
```

Comment:
I have a PR #12388 that changes the analyze API to return `ConnectorAnalyzeMetadata`. We should make this new method use the new return type. cc @findepi